grr/core/grr_response_core/lib/casing_test.py

```diff
@@ -0,0 +1,32 @@
+# Lint as: python3
+"""Tests for casing functions."""
+
+from grr_response_core.lib import casing
+
+from grr.test_lib import test_lib
+
+
+class CasingTests(test_lib.GRRBaseTest):
+  def testSnakeToCamelWorksOnEmptyString(self):
+    self.assertEqual("", casing.SnakeToCamel(""))
+
+  def testSnakeToCamelWorksOnOneWordStrings(self):
+    self.assertEqual("abcd", casing.SnakeToCamel("abcd"))
+    self.assertEqual("a", casing.SnakeToCamel("a"))
+
+  def testSnakeToCamelWorksOnRegularStrings(self):
+    self.assertEqual("thisIsACamel", casing.SnakeToCamel("this_is_a_camel"))
+    self.assertEqual("aCamelThisIs", casing.SnakeToCamel("a_camel_this_is"))
+    self.assertEqual("aBCD", casing.SnakeToCamel("a_b_c_d"))
+
```

Comment:
I would also test scenarios on strings with:

  * double underscores
  * underscore prefixes and suffixes
  * only underscores
  * uppercase characters
  * unicode characters

Reply:
Thank you for this suggestion! I implemented the new tests.